Windoors can cycle like airlocks

cacogen opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

- Can set cycle ID in airlock electronics or with mapping multi-cyclelink helper to make windoors cycle with windoors and airlocks
- Can use mapping cyclelink helper on windoors to make them cycle with another windoor or airlock (including windoors on the same tile, e.g. HoP desk)
- Windoors created by the RCD no longer produce electronics with blank settings when deconstructed
- Windoors and airlocks that are spawned rather than built no longer produce electronics with blank settings when deconstructed
- Cuts down on duplicated code for setting up door access in windoor_assembly.dm, door_assembly.dm and floor.dm
- Moves comments in door.dm to be autodoc comments, adds new comments, clarifies old ones
- Moves comments in airlock.dm to be autodoc comments
- Renames a bunch of var and proc names around airlock cycling to be more intuitive
- Adds descriptions to airlock cycle helpers to help mappers understand how they work

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

- Setting the cycle ID in airlock electronics for windoors will actually do something
- Mappers can make windoors cycle like and with airlocks
- Windoors created by the RCD will have their settings preserved in their electronics
- Windoors and airlocks that are spawned rather than built will have their settings preserved in their electronics
- Cuts down on a lot of lines of needlessly duplicated code

I've tested everything this touches and it works fine.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
expansion: Windoors can cycle like and with airlocks now
fix: Windoors created by RCDs no longer produce airlock electronics with blank settings when deconstructed
fix: Spawned airlocks no longer produce blank airlock electronics when deconstructed
code: Removes a lot of needlessly duplicated code for setting airlock access related stuff
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

cacogen committed:
Windoors can cycle like airlocks

-  Can set cycle ID in airlock electronics to make windoors cycle with other windoors and airlocks
- Windoors created by the RCD no longer produce electronics with blank settings when deconstructed
- Cuts down on duplicated code for setting up door access in windoor_assembly.dm, door_assembly.dm  and floor.dm

cacogen committed:
Remove errant debug message

bruhlookatthisdood commented:
Will there be like a mapping helper as there is for the airlock?

SuperNovaa41 commented on code/game/machinery/doors/door.dm:
```diff
@@ -352,6 +414,9 @@
+	if(delayed_close_requested)
+		delayed_close_requested = FALSE
+		addtimer(CALLBACK(src, .proc/close), 1)
 	return 1
```
why not change this one to TRUE aswell

cacogen committed:
Makes cyclelinking windoors work too

cacogen committed:
Bunch of changes

- Creates create_electronics_from_door() proc which is called in Initialize() to generate electronics for a door if it's been spawned
- Adds uses_electronics var for doors to make sure only windoors and airlocks are generated electronics on Initalize()
- Adds constructed argument to door Initialize(), used to prevent constructed doors being generated electronics when they should be passed to it instead
- Renames cyclelinkeddir var to close_other_dir as it's easier to understand
- Renames close_others_ID var to closeOtherId as it's easier to read and more intuitive
- Renames setup_cycle_link() proc to find_close_other() because it's easier to understand
- Renames cyclelinkeddoor var to close_other as it's easier to understand
- Renames update_other_id() proc to create_close_others_list() as it's more intuitive
- Removes try_close_linked_airlock() in favour of putting both methods of closing other doors under try_close_others()
- Adds and clarifies var comments to door.dm, moves them to autodoc comments
- Makes find_close_other() look for a door on the same turf first, in the case of wanting to cycle windoors on the same turf
- Makes windoors QDEL_NULL their electronics in Destroy()
- Renames cyclelink_helpers mapping helpers to be more intuitively named, and adds descriptions to clarify how they work

cacogen committed:
Moves appropriate comments to autodoc

derekhearst commented:
Can you also make windoors open with shift click as an ai and Borg? Having it be on normal click is not very intuitive.
